### ISSUE TITLE
The payment sheet does not close when timeout if user lock and unlock the mobile devices

### ISSUE THREAD

**Issue by hongduclam** [stripe/stripe-react-native#1712]
**Describe the bug**
I have added a timeout value for presentPaymentSheet({ timeout: 900s });. When a user opens the payment sheet, locks the screen, and waits for a period exceeding the timeout value, then unlocks the screen, the payment sheet is still available, and the user can proceed with the payment. This behavior violates our business rule. The timeout works correctly when the app is running in the background, but I'm unsure why there's a difference in behavior when the screen is locked. 
Im using     "@stripe/stripe-react-native": "0.37.2"
**To Reproduce**
1. Set timeout for presentPaymentSheet({ timeout: 900s });
2. Open payment sheet
3. Lock the screen
4. Waiting for after timeout
5. Unlock the screen 
6. The payment sheet still not closed

**Expected behavior**
The payment sheet should be closed after timeout when use lock the screen

**Smartphone (please complete the following information):**
 - Device: Iphone, Android
 - OS: IOS, Android

**Comment by Krunal-K-SimformSolutions** [comment]
@joyceqin-stripe: yes, i have used "@stripe/stripe-react-native": "0.51.0" version
